catlas/validation.py

```diff
@@ -0,0 +1,203 @@
+        error(
+            field,
+            "invalid elements: [%s]"
+            % ", ".join(['"%s"' % e for e in invalid_elements]),
+        )
+
+
+def validate_path_exists(field, value, error):
+    if not os.path.exists(value):
+        error(field, "file path does not exist: '%s'" % value)
+
+
+def validate_folder_exists(field, value, error):
+    """A more permissive check to check if a file can be created if it doesn't exist"""
+    path_list = value.split("/")
+    value = "/".join(path_list[:-1])
+    if not os.path.exists(value):
+        error(field, "file path's enclosing folder does not exist: '%s'" % value)
+
+
+config_schema = {
+    "memory_cache_location": {"type": "string", "check_with": validate_folder_exists},
+    "input_options": {
+        "required": True,
+        "type": "dict",
+        "schema": {
+            "adsorbate_file": {
+                "required": False,
+                "type": "string",
+                "check_with": validate_path_exists,
+            },
+            "bulk_file": {
+                "required": True,
+                "type": "string",
+                "check_with": validate_path_exists,
+            },
+        },
+    },
+    "adsorbate_filters": {
+        "type": "dict",
+        "schema": {"filter_by_smiles": {"type": "list", "schema": {"type": "string"}}},
+    },
+    "bulk_filters": {
+        "type": "dict",
+        "schema": {
+            "filter_by_bulk_ids": {"type": "list", "regex": mpid_regex},
+            "filter_ignore_mpids": {"type": "list", "regex": mpid_regex},
+            "filter_by_acceptable_elements": {
+                "type": "list",
+                "check_with": validate_element,
+            },
+            "filter_by_required_elements": {
+                "type": "list",
+                "check_with": validate_element,
+            },
+            "filter_by_num_elements": {"type": "integer"},
+            "filter_by_object_size": {"type": "integer"},
+            "filter_by_elements_active_host": {
+                "type": "dict",
+                "schema": {
+                    "active": {"type": "string", "check_with": validate_element},
+                    "host": {"type": "string", "check_with": validate_element},
+                },
+            },
+            "filter_by_bulk_e_above_hull": {
+                "type": "float",
+            },
+            "filter_by_bulk_band_gap": {"type": "float"},
+            "filter_by_element_groups": {
+                "type": "list",
+                "allowed": valid_element_groups,
+            },
+            "filter_by_pourbaix_stability": {
+                "type": "dict",
+                "schema": {
+                    "lmdb_path": {
+                        "required": True,
+                        "type": "string",
+                        # "check_with": validate_folder_exists,
+                    },
+                    "conditions_list": {
+                        "required": True,
```

Comment:
maybe I am misunderstanding but does this being required make the range option fail?

Reply:
If two things exclude each other it acts like XOR